custom_components/ttlock/services.py

```diff
@@ -98,6 +111,26 @@ def _get_coordinators(self, call: ServiceCall) -> list[LockUpdateCoordinator]:
             ]
         return []
 
+    async def handle_list_passcodes(self, call: ServiceCall) -> ServiceResponse:
+        """List all passcodes for the given entities."""
+        passcodes = {}
+
+        for coordinator in self._get_coordinators(call):
+            codes = await coordinator.api.list_passcodes(coordinator.lock_id)
+            passcodes[coordinator.data.name] = [
```

Comment:
It's probably not a big deal, but the name of the lock has no uniqueness guarantees, so I'm guessing this could have unexpected results if there are conflicts.

It might be worth a small refactor to `_get_coordinators` so that it returns a dict instead of a list, mapping the input entity id to a coordinator, then this could return the results by the requested entity id?